src/test/java/org/broadinstitute/hellbender/tools/picard/sam/AddOrReplaceReadGroupsIntegrationTest.java

```diff
@@ -0,0 +1,41 @@
+package org.broadinstitute.hellbender.tools.picard.sam;
+
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.utils.read.SamAssertionUtils;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public final class AddOrReplaceReadGroupsIntegrationTest extends CommandLineProgramTest {
+    private static final File TEST_DATA_DIR = getTestDataDir();
+    private static final File SAM_FILE = new File(TEST_DATA_DIR, "picard/sam/AddOrReplaceReadGroups/genomic_sorted_5_plus.sam");
+
+    public String getTestedClassName() {
+        return AddOrReplaceReadGroups.class.getSimpleName();
+    }
+
+    @Test
+    public void testAddCommentsToBam() throws Exception {
+        final File outputFile = BaseTest.createTempFile("AddOrReplaceReadGroups", ".sam");
+        runIt(SAM_FILE, outputFile);
+        final File expectedOutBam = new File(TEST_DATA_DIR, "picard/sam/AddOrReplaceReadGroups/genomic_sorted_5_plus.result.sam");
```

Comment:
Could you annotate with the picard version that generated this file like you did in the other.  I think that's a really good idea and one we should start doing.